.ci/azure-pipelines.yml

```diff
@@ -55,8 +68,59 @@ jobs:
       inputs:
         versionSpec: '10.x'
 
-    - script: 'yarn install'
+    - task: Cache@2
+      displayName: 'Check Cache'
+      inputs:
+        key: 'yarn | yarn.lock'
+        path: 'node_modules'
+        cacheHitVar: CACHE_RESTORED
+
+    - script: 'yarn install --frozen-lockfile'
       displayName: 'Install Dependencies'
+      condition: ne(variables.CACHE_RESTORED, 'true')
 
     - script: 'yarn run lint'
       displayName: 'Run ESLint'
+
+  - job: test
+    displayName: 'Tests'
+
+    pool:
+      vmImage: 'ubuntu-latest'
+
+    steps:
+    - task: NodeTool@0
+      displayName: 'Install Node'
+      inputs:
+        versionSpec: '10.x'
+
+    - task: Cache@2
+      displayName: 'Check Cache'
+      inputs:
+        key: 'yarn | yarn.lock'
+        path: 'node_modules'
+        cacheHitVar: CACHE_RESTORED
+
+    - script: 'yarn install --frozen-lockfile'
+      displayName: 'Install Dependencies'
+      condition: ne(variables.CACHE_RESTORED, 'true')
+
+    - script: 'yarn run jest --ci --reporters=default --reporters=jest-junit --coverage'
+      displayName: 'Run Jest'
+
+    - task: PublishTestResults@2
+      displayName: 'Publish test results'
+      condition: succeededOrFailed()
+      inputs:
+        testRunner: JUnit
+        testResultsFiles: 'junit.xml'
+
+    - task: PublishCodeCoverageResults@1
+      displayName: 'Publish coverage results'
```

Comment:
All our display names use title casing :stuck_out_tongue: and why add `succeededOrFailed()` as a condition? That sounds like it will always run, although I might be wrong.

Reply:
I'll change the casing :package: 

The PublishOrFail() was used in the Azure Pipelines docs, so I considered it best practice.

Reply:
We haven't used that in any of the other pipelines but I suppose we can leave it and wait to see what's more common.

Reply:
It makes sense to have it, has a failure in reporting tests would mean there's an issue somewhere.